frontend/src/pages/DataCatalog/dataset_explorer/DatasetExplorer.tsx

```diff
@@ -88,29 +75,37 @@ function DatasetExplorer(props: { preFilterDatasetIds: string[] }) {
     <div className={styles.DatasetExplorer}>
       <div className={styles.DatasetList}>
         <WithMetadata>
-          {(metadata) => {
-            const viewingSubsetOfDatasets =
-              getFilteredDatasetIds(metadata, activeFilter).length === 1;
+          {(datasetMetadata) => {
+            const filteredDatasets = getFilteredSources(
+              DataSourceMetadataMap,
+              activeFilter
+            );
+            // Check if more than the default filters are enabled to see if you're viewing subset of sources
+            const viewingSubsetOfSources =
+              Object.keys(activeFilter).length > 1 ||
+              activeFilter[NAME_FILTER_ID].length > 0;
 
             const defaultDatasetNames = props.preFilterDatasetIds
-              .filter((datasetId) => !!metadata[datasetId])
-              .map((datasetId) => metadata[datasetId].name);
+              .filter((datasetId) => !!FakeDatasetMetadataMap[datasetId])
```

Comment:
I don't know if we should be using this directly. Eventually we will need to be getting last-updated metadata from the server because it's dynamic, and this will make it harder to do that.

Alternatively, maybe we can split this into two: a DatasetMetadataConfig and DynamicDatasetMetadata (names need improvement lol). The former can be static config, the latter can be only the dynamic stuff that needs to come from the server.

Reply:
oh wait i actually had gotten the data properly dynamically I just forgot to use it :P

Reply:
haha nice